modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/credential/CredentialCenter.java

```diff
@@ -319,4 +326,35 @@ private boolean isAnyMandatoryCredentialVariableMissing(
                         && Objects.isNull(credentialVariable.getValue()));
     }
 
+    private void checkNullParamAndFillValueFromEnv(AbstractCredentialInfo abstractCredentialInfo) {
+        if (abstractCredentialInfo.getType().equals(CredentialType.VARIABLES)) {
+            CredentialVariables credentialVariables =
+                    (CredentialVariables) abstractCredentialInfo;
+            for (CredentialVariable variable : credentialVariables.getVariables()) {
+                if (Objects.isNull(variable.getValue())
+                        || Objects.equals(variable.getValue(), "null")) {
+                    variable.setValue(System.getenv(variable.getName()));
+                }
+            }
+        }
+    }
+
+    private void addCredentialInfoFromEnv(Csp csp,
+                                          List<AbstractCredentialInfo> abstractCredentialInfos) {
+        AbstractCredentialInfo credentialInfoFromEnv = findCredentialInfoFromEnv(csp);
```

Comment:
findCredentialInfoFromEnv - can return null. 
So we need a check here to add only non-null values here.

Reply:
ok ok, I made an oversight, it's fixed.